streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreFactory.java

```diff
@@ -66,6 +66,10 @@ default void configure(final StreamsConfig config) {
 
     boolean isVersionedStore();
 
+    // TODO: consider moving all the log configuration code (InternalTopicConfig)
```

Comment:
this would be a great cleanup, I'm definitely not a fan of the `#isWindowStore` and `retentionPeriod` logic which kind of breaks the abstraction. 

Just wondering, would we also be able to get rid of the `#historyRetention` and `#isVersionedStore` or is that used for something else?

Reply:
Btw unless you plan on doing it yourself in an immediate followup PR, please remove the TODO and file a JIRA ticket for this. As a general AK practice we don't leave inline TODOs unless they help to understand why we are currently doing something in a particular way (ie if it also makes sense as a non-TODO code comment then it's probably a good TODO as well)

Reply:
> Just wondering, would we also be able to get rid of the #historyRetention and #isVersionedStore or is that used for something else?

we would with a little bit more abstraction/refactoring. they are used somewhere to check configurations but I think it makes sense to push that into this class as well and get rid of that too.

> Btw unless you plan on doing it yourself in an immediate followup PR

I am happy to do that, I checked the code and found quite a few TODOs so I assumed it was OK. I'll just remove the TODO and file a JIRA.